File: pyhton64/nivel2/exer15.py
n1 = float(input('Nota 1: '))
n2 = float(input('Nota 2: '))
n3 = float(input('Nota 3: '))
n4 = float(input('Nota 4: '))

media = (n1 + n2 + n3 + n4) / 4

if media >= 7: 
    print('Aluno aprovado')
else:
    n_r = float(input('Nota de recuperação: '))
    nova_media = (media + n_r) / 2
    if nova_media >= 7:
        print('Aluno aprovado na recuperação')
    else:
        print(f'Aluno não foi aprovado na recuperação.\n Média = {nova_media}')

File: java64/nivel2/exer15.java
import javax.swing.JOptionPane;

public class exer15 {
    public static void main(String[] args) {
        double nota1, nota2, nota3, nota4, notaRecuperacao, media;;

        nota1 = Double.parseDouble(JOptionPane.showInputDialog(null, "Nota 1"));
        nota2 = Double.parseDouble(JOptionPane.showInputDialog(null, "Nota 2"));
        nota3 = Double.parseDouble(JOptionPane.showInputDialog(null, "Nota 3"));
        nota4 = Double.parseDouble(JOptionPane.showInputDialog(null, "Nota 4"));

        media = (nota1 + nota2 + nota3 + nota4) / 4;

        if (media >= 7) {
            JOptionPane.showMessageDialog(null, "Aprovado sem recuperação");
        } else if (media < 7) {
            notaRecuperacao = Double.parseDouble(JOptionPane.showInputDialog(null, "O aluno ficou de recuperação, digite a nota: "));
            notaRecuperacao = (notaRecuperacao + media) / 2;
            if (notaRecuperacao >= 7) {
                JOptionPane.showMessageDialog(null, "Aprovado com recuperação");
            } else {
                JOptionPane.showMessageDialog(null, "Aluno reprovado");
            }
        }
    }
}